### PR TITLE
Refactor volunteers controller

### PR DESCRIPTION
**Pull request by vidoseaver**
## Overview
Refactor the volunteers' controller

## Details
- Change one letter variables to full words
- Change abbreviated variables to multi-word variables
- Space out enumerables for easier readin'
- Replace or and and with || and &&
- Change instance variable in show for volunteers

### CODE REVIEW

**Comment by rylanb** [comment]
Nice work @vidoseaver! I did make several updates to clean-up and simplify code. It would be worth looking over and seeing if I made any mistakes and seeing cleaner ways to do some of these things. 

Updates mostly to the original code not anything you changed.

File: app/controllers/volunteers_controller.rb [hunk 1]
```diff
@@ -84,32 +84,24 @@ def index
...
-    @action = 'create'
     @regions = Region.all
-    @my_admin_regions = if current_volunteer.super_admin?
-                          @regions
-                        else
-                          current_volunteer.assignments.collect do |assingment|
-                             assingment.admin ? assingment.region : nil
-                          end.compact
-                        end
+    @my_admin_regions = current_volunteer.admin_regions
```
Comment (vidoseaver) [inline app/controllers/volunteers_controller.rb]: nice

File: app/controllers/volunteers_controller.rb [hunk 2]
```diff
@@ -129,14 +121,11 @@ def create
...
     if @volunteer.save
       flash[:notice] = 'Created successfully.'
-      unless session[:my_return_to].nil?
-        redirect_to(session[:my_return_to])
-      else
-        index
-      end
+      redirect_to(session[:my_return_to] || index)
     else
       flash[:error] = "Didn't save successfully :(. #{@volunteer.errors.full_messages.to_sentence}"
```
Comment (vidoseaver) [inline app/controllers/volunteers_controller.rb]: as per our conversation about explicit returning after redirects do you need one right here? won't it try to render after its redirected?

File: app/views/volunteers/new.html.erb [hunk 3]
```diff
@@ -1,2 +1,2 @@
 <h2>New Volunteer Account</h2>
-<%= render :partial => "form" %>
+<%= render partial: "form" %>
```
Comment (vidoseaver) [inline app/views/volunteers/new.html.erb]: dat new syntax

File: app/controllers/volunteers_controller.rb [hunk 4]
```diff
@@ -165,10 +165,11 @@ def update
...
     volunteer = Volunteer.find(params[:volunteer_id].to_i)
-    volunteer_region_ids = volunteer.regions.collect{ |region| region.id }
-    admin_region_ids = current_volunteer.assignments.collect do
-       |assignment| assignment.admin ? assignment.region.id : nil
-    end.compact
+    volunteer_region_ids = volunteer.regions.pluck(:id)
+    admin_region_ids =
+      current_volunteer.assignments.collect do |assignment|
+        assignment.admin ? assignment.region.id : nil
+      end.compact
```
Comment (vidoseaver) [inline app/controllers/volunteers_controller.rb]: Yes, this is better

**Comment by vidoseaver** [comment]
👍

File: app/controllers/volunteers_controller.rb (hunk 2 above)
Comment (rylanb) [inline app/controllers/volunteers_controller.rb]: This is the end of the codepath, so redirect_to is in the if and then in the else is the render new. 

This is also a call to the index method in the `|| index` not a redirect to index.

I had to actually fix this to be this after I merged to master as it failed a test:

`session[:my_return_to].present? ? redirect_to(session[:my_return_to]) : index `

Does that make sense? Its not super straightforward, but you need an explicit return if there is other code in the block that could still be executed.

File: app/controllers/volunteers_controller.rb (hunk 4 above)
Comment (rylanb) [inline app/controllers/volunteers_controller.rb]: @vidoseaver I actually found this in volunteer.rb: 

```
def admin_region_ids(strict = false)
    admin_regions(strict).pluck(:id)
  end

  def admin_regions(strict = false)
    if super_admin? && !strict
      Region.all
    else
      assignments.
        eager_load(:region).
        where(admin: true).
        collect(&:region)
    end
  end
```

So we could even clean these calls up further.

**Pull request merged**